sound/soc/sof/trace.c

```diff
@@ -43,6 +43,12 @@ static size_t sof_wait_trace_avail(struct snd_sof_dev *sdev,
 	if (ret)
 		return ret;
 
+	/* tracing has ended, signal EOF to client */
+	if (!sdev->dtrace_is_enabled && sdev->dtrace_eof) {
+		sdev->dtrace_eof = 0;
```

Comment:
'Signal EOF' by settings dtrace_eof to zero? That sounds counter-intuitive. Usually state_xyz = 0 means you are not in that xyz state.

Reply:
@plbossart EOF is signalled by returning 0 available bytes to read and waking up user-space. 'dtrace_eof' is internal bookkeeping so that we only signal once (otherwise an older sof-logger binary would go to a 100% CPU busy-loop after this patch is merged).